packages/client/src/data_commons_client.ts

```diff
@@ -372,9 +391,52 @@ class DataCommonsClient {
       populationObservations
     );
 
+    // Filter by start/end date if specified
+    if (params.startDate || params.endDate) {
+      return Promise.resolve(
+        dataRows.filter((dataRow) => {
+          const observationDate = dataRow.variable.observation.date;
+          if (!observationDate) {
+            return false;
+          }
+          if (params.startDate && params.startDate > observationDate) {
+            return false;
+          }
+          if (params.endDate && params.endDate < observationDate) {
+            return false;
+          }
+          return true;
+        })
+      );
+    }
+
     return Promise.resolve(dataRows);
   }
 
+  /**
+   * Overrides any of the passed in facets with values from statMetadataOverride
```

Comment:
what is the statMetadataOverride you are referring to here? I tried searching the term in this file but could not find it, maybe update the comment

Reply:
Updated